src/util/math.coffee

```diff
@@ -49,7 +49,7 @@ module.exports.successor = successor = (val) ->
       #not bothering with the max float test because javascript does not handle floats at the level
       #very well
       val + MIN_FLOAT_PRECISION_VALUE
-  else if val instanceof DateTime
+  else if val instanceof DateTime || val?.isDate
```

Comment:
Should there be a similar case for the max Time value?

Reply:
The Time values are converted to DateTime objects before they get to this point.